File: src/com/wdeanmedical/gp/entity/form/Consent.java
package com.wdeanmedical.gp.entity.form;

import java.io.Serializable;
import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Inheritance;
import javax.persistence.InheritanceType;
import javax.persistence.Table;

import com.wdeanmedical.entity.form.WDMForm;

@Entity
@Table(name = "consent_form")
@Inheritance(strategy = InheritanceType.JOINED)
public class Consent extends WDMForm implements Serializable {
  
  private static final long serialVersionUID = -319647803868784657L;
  public static String[] PHI_FIELDS = new String[] {};
  public static final String NAME = "consent";
  
  private String clientName;
  private Date dob;
  private Boolean signed;
  private String respName;
  private Date respDate;
  private String clientRel;
  private String witness;
  private Date witnessDate;
  private String initials0;
  private String initials1;
  private String initials2;
  private String initials3;
  private String initials4;
  private String initials5;
  private String initials6;
  private String initials7;
  private String initials8;
  private String initials9;
  private String initials10;
  private String initials11;
  private String initials12;
  
  
  public Consent() { 
    this.name = Consent.NAME;
  }
  
  @Column(name = "client_name")
  public String getClientName() { return clientName; }
  public void setClientName(String clientName) { this.clientName = clientName; }
  
  @Column(name = "dob")
  public Date getDob() { return dob; }
  public void setDob(Date dob) { this.dob = dob; }
  
  @Column(name = "signed")
  public Boolean getSigned() { return signed; }
  public void setSigned(Boolean signed) { this.signed = signed; }
  
  @Column(name = "resp_name")
  public String getRespName() { return respName; }
  public void setRespName(String respName) { this.respName = respName; }
  
  @Column(name = "resp_date")
  public Date getRespDate() { return respDate; }
  public void setRespDate(Date respDate) { this.respDate = respDate; }
  
  @Column(name = "client_rel")
  public String getClientRel() { return clientRel; }
  public void setClientRel(String clientRel) { this.clientRel = clientRel; }
  
  @Column(name = "witness")
  public String getWitness() { return witness; }
  public void setWitness(String witness) { this.witness = witness; }
  
  @Column(name = "witness_date")
  public Date getWitnessDate() { return witnessDate; }
  public void setWitnessDate(Date witnessDate) { this.witnessDate = witnessDate; }
  
  @Column(name = "initials0")
  public String getInitials0() { return initials0; }
  public void setInitials0(String initials0) { this.initials0 = initials0; }

  @Column(name = "initials1")
  public String getInitials1() { return initials1; }
  public void setInitials1(String initials1) { this.initials1 = initials1; }
  
  @Column(name = "initials2")
  public String getInitials2() { return initials2; }
  public void setInitials2(String initials2) { this.initials2 = initials2; }
  
  @Column(name = "initials3")
  public String getInitials3() { return initials3; }
  public void setInitials3(String initials3) { this.initials3 = initials3; }
  
  @Column(name = "initials4")
  public String getInitials4() { return initials4; }
  public void setInitials4(String initials4) { this.initials4 = initials4; }

  @Column(name = "initials5")
  public String getInitials5() { return initials5; }
  public void setInitials5(String initials5) { this.initials5 = initials5; }

  @Column(name = "initials6")
  public String getInitials6() { return initials6; }
  public void setInitials6(String initials6) { this.initials6 = initials6; }

  @Column(name = "initials7")
  public String getInitials7() { return initials7; }
  public void setInitials7(String initials7) { this.initials7 = initials7; }

  @Column(name = "initials8")
  public String getInitials8() { return initials8; }
  public void setInitials8(String initials8) { this.initials8 = initials8; }

  @Column(name = "initials9")
  public String getInitials9() { return initials9; }
  public void setInitials9(String initials9) { this.initials9 = initials9; }

  @Column(name = "initials10")
  public String getInitials10() { return initials10; }
  public void setInitials10(String initials10) { this.initials10 = initials10; }

  @Column(name = "initials11")
  public String getInitials11() { return initials11; }
  public void setInitials11(String initials11) { this.initials11 = initials11; }

  @Column(name = "initials12")
  public String getInitials12() { return initials12; }
  public void setInitials12(String initials12) { this.initials12 = initials12; }
  
  }

File: web/js/form/loaders/consent.js
function app_loadConsentForm(form) {
  var data;
  var className;
  
  if (form && form.data) {
    data = form.data;
    className = form.className.replace("com.wdeanmedical.","").replaceAll(".","-");
    form_addForm(className, data, '#consent-form');
  }
  else {
    var data = app_form;
    className = data.practiceForm.className.replace("com.wdeanmedical.","").replaceAll(".","-");
    form_addForm(className, data, '#consent-form');
  }
  
  util_debug('in app_loadConsentForm()');
  $('#consent-client-name').val(data.clientName);
  $('#consent-initials-0').val(data.initials0);
  $('#consent-initials-1').val(data.initials1);
  $('#consent-initials-2').val(data.initials2);
  $('#consent-initials-3').val(data.initials3);
  $('#consent-initials-4').val(data.initials4);
  $('#consent-initials-5').val(data.initials5);
  $('#consent-initials-6').val(data.initials6);
  $('#consent-initials-7').val(data.initials7);
  $('#consent-initials-8').val(data.initials8);
  $('#consent-initials-9').val(data.initials9);
  $('#consent-initials-10').val(data.initials10);
  $('#consent-initials-11').val(data.initials11);
  $('#consent-initials-12').val(data.initials12);
  $('#consent-dob').mask("99/99/9999");
  if(data.dob != null) { $('#consent-dob').val(dateFormat(data.dob, 'mm/dd/yyyy')); }
  $('input[name=consent-signed]').prop("checked", data.signed);
  $('#consent-resp-name').val(data.respName);
  $('#consent-resp-date').mask("99/99/9999");
  if(data.respDate != null) { $('#consent-resp-date').val(dateFormat(data.respDate, 'mm/dd/yyyy')); }
  $('#consent-client-rel').val(data.clientRel);
  $('#consent-witness').val(data.witness);
  $('#consent-witness-date').mask("99/99/9999");
  if(data.witnessDate != null) { $('#consent-witness-date').val(dateFormat(data.witnessDate, 'mm/dd/yyyy')); }
}
